app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -74,46 +75,186 @@ class RefreshController @Inject constructor(
         stopped: (() -> Unit)? = null,
     ): Pair<Set<Thread>?, Throwable?> {
 
-        suspend fun refreshWithRunCatching(job: Job, isFirstTime: Boolean = true): Pair<Set<Thread>?, Throwable?> = runCatching {
-            withContext(Dispatchers.IO + job) {
-                if (isFirstTime) {
-                    started?.invoke()
-                } else {
-                    delay(Utils.DELAY_BEFORE_FETCHING_ACTIVITIES_AGAIN)
-                    ensureActive()
+        SentryLog.i("API", "Refresh threads with mode: $refreshMode | (${folder.name})")
+
+        endOfMessagesReached = false
+
+        refreshThreadsJob?.cancel()
+        refreshThreadsJob = Job()
+
+        return refreshWithRunCatching(refreshThreadsJob!!, refreshMode, mailbox, folder, okHttpClient, realm, started, stopped)
+            .also {
+                it.first?.let {
```

Comment:
```suggestion
                threads?.let {
```
or 

```suggestion
                if (threads != null)
```

Because using a let without using the copy of the value seems less readable than a if

Reply:
@sirambd & @JorisBodin will refuse: _"It's Kotlin not Java !"_